kolibri/plugins/coach/assets/src/composables/useQuizCreation.js

```diff
@@ -436,6 +437,67 @@ export default function useQuizCreation() {
     }
   });
 
+  /**
+   * Map of exercise id to array of question items that are not used for each exercise
+   * @type {ComputedRef<Object.<string, string[]>>}
+   */
+  const activeExercisesUnusedQuestionsMap = computed(() => {
+    const map = {};
+    for (const exercise of Object.values(get(activeResourceMap))) {
+      const unusedQuestions = exercise.assessmentmetadata.assessment_item_ids
+        .map(aid => `${exercise.id}:${aid}`)
+        .filter(qid => !get(allQuestionsInQuiz).find(q => q.item === qid));
+      map[exercise.id] = unusedQuestions;
+    }
+    return map;
+  });
+
+  /**
+   * Method to replace questions in `questionItems` with new questions selected from
+   * the unused questions of the exercises that each question belongs to.
+   * @param {Array<string>} questionItems
+   * @throws {Error} If there are no enough unused questions in the exercise to replace a question
+   */
+  function autoReplaceQuestions(questionItems = []) {
+    if (!questionItems?.length) {
+      return;
+    }
+    const questionsToReplace = questionItems
+      .map(questionItem => get(activeQuestions).find(q => q.item === questionItem))
+      .filter(Boolean);
+    const exercises = uniq(questionsToReplace.map(q => q.exercise_id));
+
+    const shuffledExercisesUnusedQuestionsMap = {};
+    exercises.forEach(exerciseId => {
```

Comment:
Not at all blocking - I tend to prefer for..of loops to forEach to iterate over an array - this is mostly stylistic, rather than because of the very mild performance penalty that comes from the redefinition of the anonymous function each time this code is executed - as it is negligible in most cases!

Reply:
Thanks @rtibbles! Didnt know what was the preferred way, will have that in mind in the future!

Reply:
Well, I said _I_ very specifically - I don't think we've made a decision as a team :)